mcp/mcp_server_test.go

```
@@ -0,0 +1,117 @@
+		}
+
+		result2, err2 := client.CallTool(context.Background(), request)
+		if err2 != nil {
+			t.Errorf("CallTool failed: %v", err2)
+		}
+
+		if result2 == nil {
+			t.Errorf("Expected a result")
+		}
+
+		if result2.IsError {
```

golangci-lint (1.64.8)

SA5011: possible nil pointer dereference

(staticcheck)